Makefile

```diff
@@ -73,10 +73,11 @@ $(GO_APIDIFF): $(TOOLS_DIR)/go.mod # Build go-apidiff from tools folder.
 $(CONTROLLER_GEN): $(TOOLS_DIR)/go.mod # Build controller-gen from tools folder.
 	cd $(TOOLS_DIR) && go build -tags=tools -o bin/controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen
 
-$(GOLANGCI_LINT): .github/workflows/golangci-lint.yml # Download golanci-lint using hack script into tools folder.
-	hack/ensure-golangci-lint.sh \
-		-b $(TOOLS_BIN_DIR) \
-		$(shell cat .github/workflows/golangci-lint.yml | grep "version: v" | sed 's/.*version: //')
+$(GOLANGCI_LINT): .github/workflows/golangci-lint.yml # Download golangci-lint using hack script into tools folder.
+	GOBIN=$(abspath $(TOOLS_BIN_DIR)) go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(shell cat .github/workflows/golangci-lint.yml | grep "version: v" | sed 's/.*version: //')
+
+.PHONY: tools
+tools: $(GO_APIDIFF) $(CONTROLLER_GEN) $(GOLANGCI_LINT)
```

Comment:
why are we diverting here from upstream?

Reply:
I carried these changes over from the kcp-1.26 branch, where the commits just said

> [UPSTREAM: <carry>: make: use go install for lint](https://github.com/kcp-dev/controller-runtime/commit/5aca657b62d8d0d13f7ff4dbbf1a12cbde71f516)
> [UPSTREAM: <carry>: make: add tools target](https://github.com/kcp-dev/controller-runtime/commit/58443a8197164f69754e2f51ea60ae7187dbd552)

If you want I can remove them and see if something breaks, as I am also not sure why exactly we added these.

Reply:
No strong opinion, was just curious.